### DISCUSSION TITLE
Any issues with using trilium to store copies of receipts?

### DISCUSSION THREAD

**Discussion by justyns** [TriliumNext/Trilium#4200] [Q&A]
I'm currently in the process of testing trilium and so far have been pretty impressed with it.  One of my workflows I want to move from Joplin to Trilium is storing receipts/confirmations.   E.g. if I pay a bill online, I usually use joplin's web clipper to save a copy of the confirmation number/etc.

Since I originally migrated from evernote to joplin (and then to obsidian, but without receipts), I've accumulated several thousand "notes" over the years.

Does anyone have any experience using trilium to store receipts/confirmations?  I'm wondering if having so many would cause issues when searching for actual notes, or any performance issues in general?

Note that these are mostly just text based receipts, not pdfs or images.

**Comment by jwhonce** [comment]
I have imported almost 500 recipes that are text based with some images, another 200 notes in the calendar tree and 100 tips/hacks notes. I have not seen Trilium's performance change. I know that is less than you are looking at, but thought this might help. I use a couple of the Python bindings and have moved from code walking note trees to search based traversing as the performance has been great and it makes the code easier. I use/create attributes to filter the searches.

I have laptops connected to a sync server on my LAN. 

PS. Web clipping has been a mixed bag, when it fails I get an unreadable new note.

**Reply by justyns** [reply]
Thanks @jwhonce , it is helpful to have more data points.   I've been testing importing a bunch of notes multiple times (trying to fix import issues) and ended up with probably 10,000 notes.  I didn't notice any performance issues yet!   The only slowdown seems to be during sync, but the UI itself is still responsive.

I haven't tried the web clipper yet, but it's on my list.

**Comment by zadam** [comment]
The design goal of Trilium is to work fine with 100 000 notes. I personally have > 30 000 notes and things are fast (enough).

**Reply by justyns** [reply]
Thanks!  After posting this, I did see that mentioned in the documentation somewhere.  I'm glad it's one of the goals.   Other than sync slowing down some, I haven't noticed much of an issue with importing a large number of notes so far.

Would you expect the encryption/protected subtrees to also perform OK with that many notes?   I'm wondering if it's feasible to encrypt/protect the entire trilium document, but wasn't sure if it has to load everything into memory or something else.

**Reply by zadam** [reply]
I think it should perform OK. There might be slowdown when entering the protected session, since basic note data is loaded in the backend memory, and entering the protected session would decrypt all the titles stored there. But I believe that decrypting 100 000 short strings shouldn't be a big problem on modern CPUs.

**Comment by sottey** [comment]
@zadam I appreciate the 100k goal, and I have seen no issues so far. I have one question (I could look at the code, but you may be able to answer more quickly than I could code review):

Does marking notes as "Archived" exclude them from the performance impact? Is there any impact in marking notes as Archived 
other than visual changes?

Thanks!

**Reply by zadam** [reply]
There's no real effect on the performance of archived notes, they are meant to have semantic meaning. One of the bigger effect is that archiving strongly penalizes notes in the search, autocomplete.
